logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt

```diff
@@ -80,6 +80,40 @@ class MLSConversationDataSource(
             establishMLSGroup(groupID, members)
         }
 
+    override suspend fun addMemberToMLSGroup(groupID: String, members: List<UserId>): Either<CoreFailure, Unit> =
+        //TODO: check for federated and non-federated members
+        keyPackageRepository.claimKeyPackages(members).flatMap { keyPackages ->
+            mlsClientProvider.getMLSClient().flatMap { client ->
+                val clientKeyPackageList = keyPackages
+                    .map {
+                        Pair(
+                            CryptoQualifiedClientId(it.clientID, CryptoQualifiedID(it.userId, it.domain)),
+                            it.keyPackage.decodeBase64Bytes()
+                        )
+                    }
+                client.addMember(groupID, clientKeyPackageList)?.let { (handshake, welcome) ->
+                    wrapApiRequest {
+                        mlsMessageApi.sendMessage(MLSMessageApi.Message(handshake))
+                    }.flatMap {
+                        wrapApiRequest {
+                            mlsMessageApi.sendWelcomeMessage(MLSMessageApi.WelcomeMessage(welcome))
+                        }
+                    }.flatMap {
+                        wrapStorageRequest {
+                            val list = members.map {
+                                com.wire.kalium.persistence.dao.Member(idMapper.toDaoModel(it))
+                            }
+                            conversationDAO.insertMembers(list, groupID)
```

Comment:
this will cause a crash if the Members information is  not stored in the `Users` table
i suggest not using the DAO directly and use the conversation repo (and make sure that the members info in stored in the DB)

Reply:
This class itself is a repository! So it's ok to access to the DAO layer.

Reply:
what i mean is the member have a foreign key relation with the user id in the Users table so  inserting a member directly in the table without inserting the user info in the user table first can cause a crash